### PR TITLE
Make ClusterStatsMonitor http client options can configurable

### PR DESCRIPTION
**Pull request by Alienero**
## Description
Make ClusterStatsMonitor http client options can configurable.

## Additional context and related issues
#434 

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* 
```

### CODE REVIEW

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsInfoApiMonitor.java [hunk 1]
```diff
@@ -32,10 +32,14 @@
...
 {
-    //TODO: make client options configurable
-    private static final HttpClient client = new JettyHttpClient(new HttpClientConfig());
     private static final Logger log = LoggerFactory.getLogger(ClusterStatsInfoApiMonitor.class);
     private static final JsonResponseHandler<ServerInfo> SERVER_INFO_JSON_RESPONSE_HANDLER = createJsonResponseHandler(jsonCodec(ServerInfo.class));
+    private final HttpClient client;
+
+    public ClusterStatsInfoApiMonitor(HttpClient client)
+    {
+        this.client = client;
```
Comment (oneonestar) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsInfoApiMonitor.java:41]:
```suggestion
        this.client = requireNonNull(client, "client is null");
```

**Comment by oneonestar** [comment]
Looks good.
Please sign the CLA before we can merge.
Also, adding docs would be appreciated.

**Comment by Alienero** [comment]
I have rebased the changes, added some usage documentation, and signed the CLA file

**Review by oneonestar — APPROVED** [review]

**Comment by oneonestar** [comment]
Thanks for the contribution.

**Comment by mosabua** [comment]
@oneonestar please suggest release notes entry in the description. Also next time we should try to improve commit message before or with merge.

**Pull request merged**